front/src/app/components/forms/register-agencies/RegisterAgenciesForm.tsx

```diff
@@ -1,88 +1,220 @@
 import { fr } from "@codegouvfr/react-dsfr";
 import { Button } from "@codegouvfr/react-dsfr/Button";
-import { zodResolver } from "@hookform/resolvers/zod";
-import { keys } from "ramda";
-import React from "react";
-import { ErrorNotifications } from "react-design-system";
-import { useForm } from "react-hook-form";
+import Checkbox from "@codegouvfr/react-dsfr/Checkbox";
+import Input from "@codegouvfr/react-dsfr/Input";
+import Table from "@codegouvfr/react-dsfr/Table";
+import React, { ElementRef, useState } from "react";
+import { Tag } from "react-design-system";
 import { useDispatch } from "react-redux";
-import { AgencyOption, agencyOptionSchema, domElementIds } from "shared";
-import { formAgencyFieldsLabels } from "src/app/contents/forms/agency/formAgency";
 import {
-  displayReadableError,
-  getFormContents,
-  toErrorsWithLabels,
-} from "src/app/hooks/formContents.hooks";
+  AgencyId,
+  InclusionConnectedUser,
+  agencyKindToLabelIncludingIF,
+  domElementIds,
+} from "shared";
+import { useAppSelector } from "src/app/hooks/reduxHooks";
+import { routes } from "src/app/routes/routes";
+import { agenciesSelectors } from "src/core-logic/domain/agencies/agencies.selectors";
+import { agenciesSlice } from "src/core-logic/domain/agencies/agencies.slice";
 import { inclusionConnectedSlice } from "src/core-logic/domain/inclusionConnected/inclusionConnected.slice";
-import { z } from "zod";
-import { MultipleAgencyInput } from "./MultipleAgencyInput";
 
-type WithAgenciesOptions = {
-  agencies: AgencyOption[];
-};
-
-const registerAgenciesFormSchema: z.Schema<WithAgenciesOptions> = z.object({
-  agencies: z.array(agencyOptionSchema).nonempty(),
-});
-
-export const RegisterAgenciesForm = () => {
-  const { handleSubmit, setValue, formState, getValues, watch } =
-    useForm<WithAgenciesOptions>({
-      defaultValues: {
-        agencies: [],
-      },
-      resolver: zodResolver(registerAgenciesFormSchema),
-    });
-  const { getFormErrors } = getFormContents(formAgencyFieldsLabels);
+export const RegisterAgenciesForm = ({
+  currentUser,
+}: {
+  currentUser: InclusionConnectedUser;
+}) => {
   const dispatch = useDispatch();
+  const [inputValue, setInputValue] = useState<string>("");
+  const inputElement = React.useRef<ElementRef<"input">>(null);
+
+  const [isAllChecked, setIsAllChecked] = useState<boolean>(false);
+  const [checkedAgencies, setCheckedAgencies] = useState<AgencyId[]>([]);
```

Comment:
```suggestion
  const [selectedAgencyIds, setSelectedAgencyIds] = useState<AgencyId[]>([]);
```

Checked c'est très lié à l'implem UI